### PR TITLE
feat: Responsive Style Props - Hook

### PR DESCRIPTION
**Pull request by josh-bagwell**
## Summary

Fixes: #[1503](https://github.com/Workday/canvas-kit/issues/1503) 

Create responsive props that will allow the ability to apply responsive styles to components. 

category

### Release Note

---

## Checklist

- [ ] tests are changed or added
- [ ] code has been documented
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](https://workday.github.io/canvas-kit/?path=/story/welcome-dev-docs-api-pattern-guidelines--page)

### CODE REVIEW

File: modules/react/layout/lib/utils/res.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  return matches;
+};
+
+export const useBreakpoints = () => {
+  const breakpoints = {
+    zero: useMediaQuery('(min-width: 0)'),
+    isSm: useMediaQuery('(min-width: 320px)'),
+    isMd: useMediaQuery('(min-width: 768px)'),
+    isLg: useMediaQuery('(min-width: 1024px)'),
+    isXl: useMediaQuery('(min-width: 1440px)'),
```
Comment (mannycarrera4) [inline modules/react/layout/lib/utils/res.ts:26]: Do we want to tie these in at all with our Breakpoints from theme?

File: modules/react/layout/stories/Flex.stories.mdx [hunk 2]
```diff
@@ -23,6 +24,8 @@ yarn add @workday/canvas-kit-react
 
 ## Usage
```
Comment (mannycarrera4) [inline modules/react/layout/stories/Flex.stories.mdx:26]: Can we add more context here on how to use this, maybe even a snippet of an example?

File: modules/react/layout/stories/examples/Flex/FlexCardResProps.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      fontWeight: type.properties.fontWeights.bold,
+    }}
+    {...props}
+  />
+);
+
+const Body = props => <p style={{...type.levels.body.small, margin: 0}} {...props} />;
+
+export const FlexCardResProps = () => {
+  const resStyles = useBreakpointValues({
```
Comment (mannycarrera4) [inline modules/react/layout/stories/examples/Flex/FlexCardResProps.tsx:23]: I think we should document how to use this. This example is great but a little more context might help!

**Comment by alanbsmith** [comment]
Closing this. We'll submit a new PR to resolve #1503

**Pull request closed without merging**